### ISSUE TITLE
passing a hash in to perform raises an error

### ISSUE THREAD

**Issue by willrax** [machty/ember-concurrency#99]
Hey,

I recently updated our app to Ember 2.10.0 and discovered a small bug.

Given the following button with a hash argument:

```hbs
<button onclick={{perform updateUser (hash admin=false)}}>
  Click
</button>
```

and the follow task:

```js
updateUser: task(function* (properties) {
  yield timeout(500); // represents a save  
})
```

Clicking on the button throws this error:

```
errors.js:24 TypeError: Cannot convert object to primitive value
    at Array.join (native)
    at Class.toString (http://localhost:4200/assets/vendor.js:180976:69)
    at Object.toString (http://localhost:4200/assets/vendor.js:55125:18)
    at Object.set (http://localhost:4200/assets/vendor.js:36754:131)
    at Class.set (http://localhost:4200/assets/vendor.js:49381:26)
    at Class.schedule (http://localhost:4200/assets/vendor.js:180430:12)
    at Class._perform (http://localhost:4200/assets/vendor.js:181558:23)
    at Class.perform (http://localhost:4200/assets/vendor.js:181362:30)
    at http://localhost:4200/assets/vendor.js:180310:31
    at Backburner.run (http://localhost:4200/assets/vendor.js:16154:23)
```

This seems to link to the `toString` function in `-task-instance` and specifically the `args.join`

```js
    toString: function toString() {
      var taskString = "" + this.task;
      return spliceSlice(taskString, -1, 0, '.perform(' + this.args.join(',') + ')');
    },
```

Inspecting the args gives this

<img width="200" alt="screen shot 2016-12-04 at 11 08 15" src="https://cloud.githubusercontent.com/assets/94960/20865721/01403af2-ba12-11e6-81f1-5701c2f63c2b.png">

I think it might be related to the hash helper and the way these might have changed in Ember 2.10.0 but I'm not sure what the correct way to fix this would be.

**Comment by willrax** [comment]
related to https://github.com/emberjs/ember.js/issues/14489

**Comment by rlivsey** [comment]
Just ran into this myself, "fixed" for now by moving from passing a hash to list of arguments.

**Comment by machty** [comment]
Fixed in 0.7.16
